source/base/TextBuilder.ooc

```diff
@@ -0,0 +1,140 @@
+/*
+* Copyright (C) 2015 - Simon Mika <simon@mika.se>
+*
+* This sofware is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 2.1 of the License, or (at your option) any later version.
+*
+* This software is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this software. If not, see <http://www.gnu.org/licenses/>.
+*/
+use ooc-base
+use ooc-collections
+
+TextBuilder: class {
+	_data := VectorList<Text> new(32, false)
+	count ::= this _data count
+
+	init: func ~default
+	init: func ~vector (=_data)
+	init: func ~text (value: Text) {
+		this init()
+		this append(value)
+	}
+	init: func ~cstring (value: CString, length: Int) {
+		this init()
+		this append(value, length)
+	}
+	init: func ~this (original: This) {
+		this init()
+		for (i in 0 .. original count)
+			this append(original[i] copy())
+	}
+	free: func {
+		for (i in 0 .. this count)
+				this _data[i] free(Owner Callee)
+		this _data free()
+		super()
+	}
+	copy: func -> This {
+		This new(this)
+	}
+	append: func ~cstring (value: CString, length: Int) {
+		this append(Text new(value, length))
+	}
+	append: func ~text (value: Text) {
+		this _data add(value)
+	}
+	append: func ~This (other: This) {
```

Comment:
Having no suffix when the argument is of type `This`, as you did in `Text`, seems like a good idea to me.